src/FuncSharp/Extensions/ObjectExtensions_ValueMatch.tt

```diff
@@ -36,6 +36,25 @@ public static partial class ObjectExtensions
         throw new ArgumentException("The value " + value.SafeToString() + " does not match any of the <#= i #> specified values.");
     }
 
+    /// <summary>
+    /// Matches the value with the specified parameters and returns result of the corresponding value.
+    /// </summary>
+    [Pure]
+    public static TResult Match<T, TResult>(
+        this T value,
+<#= Lines(i, j => Indent(12) + "T " + Value(j) + ", TResult f" + j, separator: ",") #>,
+        TResult otherwise = default(TResult))
+    where T: IEquatable<T>
```

Comment:
Do we need the IEquatable? Most of our stuff doesn't in fact use IEquatable. And the other methods also don't check for IEquatable.  So for example this wouldn't work for an enum.

Reply:
Yes, it covers base types, which support IEquatable, which i choose, because in dotnet you are unable to set constraint against value types. So this subset helps in some cases at least.

Reply:
What I'm saying is that you can omit the `where T :  IEquatable` and make it simply apply to all the types like the other match method does. 

Putting it there means that a lot of the places wouldn't be able to use this method.